public/app/features/alerting/unified/api/alertmanager.ts

```diff
@@ -171,28 +171,52 @@ export async function testReceivers(
     receivers,
     alert,
   };
-  const result = await lastValueFrom(
-    getBackendSrv().fetch<TestReceiversResult>({
-      method: 'POST',
-      data,
-      url: `/api/alertmanager/${getDatasourceAPIId(alertManagerSourceName)}/config/api/v1/receivers/test`,
-      showErrorAlert: false,
-      showSuccessAlert: false,
-    })
-  );
+  try {
+    const result = await lastValueFrom(
+      getBackendSrv().fetch<TestReceiversResult>({
+        method: 'POST',
+        data,
+        url: `/api/alertmanager/${getDatasourceAPIId(alertManagerSourceName)}/config/api/v1/receivers/test`,
+        showErrorAlert: false,
+        showSuccessAlert: false,
+      })
+    );
 
-  // api returns 207 if one or more receivers has failed test. Collect errors in this case
-  if (result.status === 207) {
-    throw new Error(
-      result.data.receivers
-        .flatMap((receiver) =>
-          receiver.grafana_managed_receiver_configs
-            .filter((receiver) => receiver.status === 'failed')
-            .map((receiver) => receiver.error ?? 'Unknown error.')
-        )
-        .join('; ')
+    // api returns 207 if one or more receivers has failed test. Collect errors in this case
+    if (result.status === 207) {
+      throw new Error(getReceiverResultError(result.data));
+    }
+  } catch (error) {
+    // api returns 400 if all receivers failed and 408 if all timed out
+    if (isFetchError(error) && (error.status === 400 || error.status === 408)) {
+      if (isTestReceiversResult(error.data)) {
+        throw new Error(getReceiverResultError(error.data));
+      }
+    }
+
+    throw error;
+  }
+}
+
+function isTestReceiversResult(data: any): data is TestReceiversResult {
+  if (typeof data.receivers === 'object' && Array.isArray(data.receivers)) {
```

Comment:
`Array.isArray(data.receivers)` should be sufficient here, since that implies `typeof object`